lib/Evaluate/shape.cpp

```diff
@@ -256,6 +256,13 @@ MaybeExtentExpr GetExtent(
         }
       }
     }
+  } else if (const auto *assoc{
+                 symbol.detailsIf<semantics::AssocEntityDetails>()}) {
+    if (auto shape{GetShape(context, assoc->expr())}) {
+      if (!shape->empty()) {
+        return std::move(shape->back());
```

Comment:
Doesn't this have to take into account dimension? E.g. `shape->at(dimension)`

Reply:
Right.